renku/infrastructure/storage/storage_service.py

```diff
@@ -0,0 +1,154 @@
+        expected_response=[200],
+    ):
+        """Send an unauthenticated request."""
+        request_method: Callable[..., Any]
+        if method == "GET":
+            request_method = requests.get
+        elif method == "POST":
+            request_method = requests.post
+        elif method == "PUT":
+            request_method = requests.put
+        elif method == "DELETE":
+            request_method = requests.delete
+        else:
+            raise NotImplementedError()
+
+        url = f"{self.base_url}{path}"
+        headers = None
+
+        if auth:
+            headers = self._auth_headers()
+
+        resp = request_method(url, headers=headers, params=parameters, data=body, timeout=TIMEOUT)  # type: ignore
+
+        if resp.status_code not in expected_response:
+            raise errors.RequestError(f"Request to storage service failed ({resp.status_code}): {resp.text}")
+
+        return resp.json()
+
+    def list(self, project_id: str) -> List[CloudStorageWithSensitiveFields]:
+        """List storage configured for the current project."""
+        response = self._send_request("/storage", parameters={"project_id": project_id}, auth=True)
+        results = []
+        for res in response:
+            results.append(
+                CloudStorageWithSensitiveFields(CloudStorage.from_dict(res["storage"]), res["sensitive_fields"])
+            )
+
+        return results
+
+    def create(self, storage: CloudStorage) -> CloudStorageWithSensitiveFields:
+        """Create a new cloud storage."""
+        if storage.storage_id is not None:
+            raise ValueError("Cannot create storage with 'storage_id' already set.")
+        if storage.project_id is None:
+            raise ValueError("'project_id' must be set when creating CloudStorage.")
+        response = self._send_request(
+            "/storage", body=asdict(storage), method="POST", auth=True, expected_response=[201]
+        )
+        return CloudStorageWithSensitiveFields(
+            CloudStorage.from_dict(response["storage"]), response["sensitive_fields"]
+        )
+
+    def edit(self, storage_id: str, new_storage: CloudStorage) -> CloudStorageWithSensitiveFields:
+        """Edit a cloud storage."""
+        response = self._send_request(f"/storage/{storage_id}", body=asdict(new_storage), method="PUT", auth=True)
+        return CloudStorageWithSensitiveFields(
+            CloudStorage.from_dict(response["storage"]), response["sensitive_fields"]
+        )
+
+    def delete(self, storage_id: str) -> None:
+        """Delete a cloud storage."""
+        self._send_request(f"/storage{storage_id}", method="DELETE", auth=True, expected_response=[204])
+
+    def validate(self, storage: CloudStorage) -> None:
+        """Validate a cloud storage.
+
+        Raises an exception for invalid storage.
+        """
+        self._send_request(
+            "/storage_schema/validate", body=storage.configuration, method="POST", expected_response=[204]
```

Comment:
Is the expected reponse for validate supposed to be really 204?

Reply:
yes when given a valid config it just returns an empty 204 and if invalid an appropriate 4xx code